admin/server/organizations.go

```diff
@@ -187,12 +188,25 @@ func (s *Server) AddOrganizationMember(ctx context.Context, req *adminv1.AddOrga
 		if !errors.Is(err, database.ErrNotFound) {
 			return nil, status.Error(codes.Internal, err.Error())
 		}
-		// Create phantom user
-		// TODO: Replace by an invite-based approach
-		user, err = s.admin.CreateOrUpdateUser(ctx, req.Email, "", "")
+
+		if claims.OwnerType() != auth.OwnerTypeUser {
+			return nil, status.Error(codes.Unauthenticated, "not authenticated as a user")
+		}
+		// Invite user to join the organization
+		role, err := s.admin.DB.FindOrganizationRole(ctx, req.Role)
 		if err != nil {
+			if errors.Is(err, database.ErrNotFound) {
+				return nil, status.Error(codes.InvalidArgument, "role not found")
+			}
 			return nil, status.Error(codes.Internal, err.Error())
 		}
+		err = s.admin.InviteUserToOrganization(ctx, req.Email, claims.OwnerID(), org.ID, role.ID, org.Name, role.Name)
```

Comment:
When we have `s.admin.InviteUserToOrganization` anyway, I think we should move all the logic in this function (after the claims checks) into it

Reply:
Same applies for projects

Reply:
after moving the role check above, there is nothing more left to move into this method